core/templates/dev/head/services/ExplorationHtmlFormatterService.js

```diff
@@ -23,8 +23,21 @@ oppia.factory('ExplorationHtmlFormatterService', [
   '$filter', 'extensionTagAssemblerService', 'HtmlEscaperService',
   function($filter, extensionTagAssemblerService, HtmlEscaperService) {
     return {
+      /**
+      * @param {string} interactionId - The id of interaction in camel case.
+      * @param {array} interactionCustomizationArgSpecs - The various attributes
+      * that depends on interaction.
+      * @param {boolean} canAcceptAnswers - Whether or not last answer is
+      * displayed permanently after learner has submitted an answer for
+      * non-inline interactions.
```

Comment:
Design question: is it guaranteed that, if this condition holds, there's a $scope.lastAnswer in the parent? If not, we may need to make that a requirement and specify it here.

Reply:
This is set to true only in files related to player view and in all places where the interaction is called from player view, lastAnswer is set, so currently, I don't think additional checks are necessary.

Reply:
I'm still thinking about the case where another dev, in the future, creates an interaction HTML div. What do they need to know in order to make sure everything works correctly and that they choose the correct value for canAcceptAnswers here? If they need to have a lastAnswer in their div then we should tell them that.

Reply:
Flagging this as the last remaining comment.

Reply:
In editor_tab, canAcceptAnswer should be set to false regardless of interaction as there is no lastAnswer to show. 
In any file related to exploration_player, if this function is called, this should be set to true as only then will lastAnswer be visible. Setting actual value of lastAnswer is done in both supplementalCardDirective and tutorCardDirective by calling PlayerTranscriptsService.getLastAnswerOnActiveCard(), so nothing need be added by dev related to setting the actual value of lastAnswer. Just setting canAcceptAnswer to true would suffice.

Reply:
OK, sorry, maybe I'm not being clear. Here are some examples of the sorts of things I'm thinking of.

- We may allow people to suggest changes to interactions. These suggestions would then be viewable on the feedback tab of the exploration editor. They may include a playable version of the interaction so that the creator can test the changes. Should the arg you've specified be set to true or false? Does anything need to be done with regards to ensuring that the feedback tab contains a $scope.lastAnswer variable?
- We are planning to add a training interface in the exploration editor tab. Creators can submit answers to this, see Oppia's feedback to them, and then mark that feedback as desired/not-desired. Should the arg you've specified be set to true or false? Does anything need to be done with regards to ensuring that the training modal contains a $scope.lastAnswer variable?

And so on. Basically what is important is to future-proof this function, since devs will be using it for all sorts of things, possibly beyond the contexts you've considered. So the function needs to be really clear and concrete about what the minimal requirements for using it are.

Reply:
Oh, OK.
How about this? If interaction is called in exploration_player view (including sample runs by creator and in the suggest changes section), set canAcceptAnswer to true and call $scope.lastAnswer = PlayerTranscriptService.getLastAnswerOnActiveCard(index) to actually set value of lastAnswer and in all other views (like when the author is changing the interaction in editor_tab), set canAcceptAnswer to false.

Reply:
That seems like a reasonable starting point, but you'll need to be clearer about what "in exploration_player view" means. For the suggest changes thing I wasn't talking about a literal full-blown player view, just a "testable" copy of the interaction.

Maybe you can say: If this function is called in the exploration_player view (including the preview mode), callers should ensure that canAcceptAnswer is set to true and $scope.lastAnswer = PlayerTranscriptService.getLastAnswerOnActiveCard(index) is set on the parent controller of the returned tag. Otherwise, canAcceptAnswer should be set to false.

Also, if this is what we're going with, we should change canAcceptAnswer to a better name that actually conveys the above.

Reply:
Ok, will change the description. How about canShowLastAnswer?

Reply:
Still not clear. What does it mean for the interaction to "show a last answer"?

parentMaintainsAnswerHistory, perhaps? (Is that accurate, though?)

Reply:
(Btw, finding good names is hard. Feel free to suggest alternatives.)

Reply:
How about parentRecordsLastSubmittedAnswer? as entire history of submitted answers is not used anyways.